test/issues/reparce_tag_file_placeholder.cpp

```diff
@@ -0,0 +1,139 @@
+
+//  Distributed under the Boost Software License, Version 1.0.
+//  See http://www.boost.org/LICENSE_1_0.txt
+
+//  Library home page: http://www.boost.org/libs/filesystem
+
+#include <boost/scope_exit.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+#include <iostream>
+
+# include <windows.h>
+# include <winnt.h>
+
+#ifdef _MSC_VER
+#  pragma comment(lib, "Advapi32.lib")
+#endif
+
+// Test correct boost::filesystem::status when reparse point ReparseTag set to IO_REPARSE_TAG_FILE_PLACEHOLDER
+// https://docs.microsoft.com/en-us/windows/compatibility/placeholder-files?redirectedfrom=MSDN
+
+#ifndef IO_REPARSE_TAG_FILE_PLACEHOLDER
+#  define IO_REPARSE_TAG_FILE_PLACEHOLDER (0x80000015L)
+#endif
+
+typedef struct _REPARSE_DATA_BUFFER {
+  ULONG  ReparseTag;
+  USHORT  ReparseDataLength;
+  USHORT  Reserved;
+  union {
+    struct {
+      USHORT  SubstituteNameOffset;
+      USHORT  SubstituteNameLength;
+      USHORT  PrintNameOffset;
+      USHORT  PrintNameLength;
+      ULONG  Flags;
+      WCHAR  PathBuffer[1];
+    } SymbolicLinkReparseBuffer;
+    struct {
+      USHORT  SubstituteNameOffset;
+      USHORT  SubstituteNameLength;
+      USHORT  PrintNameOffset;
+      USHORT  PrintNameLength;
+      WCHAR  PathBuffer[1];
+    } MountPointReparseBuffer;
+    struct {
+      UCHAR  DataBuffer[1];
+    } GenericReparseBuffer;
+  };
+} REPARSE_DATA_BUFFER, *PREPARSE_DATA_BUFFER;
+
+#define REPARSE_DATA_BUFFER_HEADER_SIZE \
+  FIELD_OFFSET(REPARSE_DATA_BUFFER, GenericReparseBuffer)
+
+bool obtain_restore_privilege()
+{
+  HANDLE hToken;
+  if (!OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES, &hToken))
+  {
+    std::cout << "OpenProcessToken() failed with: " << GetLastError() << std::endl;
+    return false;
+  }
+
+  BOOST_SCOPE_EXIT(&hToken)
+  {
+    CloseHandle(hToken);
+  } BOOST_SCOPE_EXIT_END;
+
+  TOKEN_PRIVILEGES tp;
+  if (!LookupPrivilegeValue(NULL, SE_RESTORE_NAME, &tp.Privileges[0].Luid))
+  {
+    std::cout << "LookupPrivilegeValue() failed with: " << GetLastError() << std::endl;
+    return false;
+  }
+
+  tp.PrivilegeCount = 1;
+  tp.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
+
+  if (!AdjustTokenPrivileges(hToken, FALSE, &tp, sizeof(TOKEN_PRIVILEGES), NULL, NULL))
+  {
+    std::cout << "AdjustTokenPrivileges() failed with: " << GetLastError() << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+bool create_io_reparse_file_placeholder(const wchar_t* name)
+{
+  if (!obtain_restore_privilege())
+  {
+    return false;
+  }
+
+  HANDLE hHandle = CreateFileW(name, GENERIC_WRITE,
+    FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, CREATE_ALWAYS,
+    FILE_FLAG_OPEN_REPARSE_POINT, 0);
+
+  if (hHandle == INVALID_HANDLE_VALUE)
+  {
+    std::cout << "CreateFile() failed with: " << GetLastError() << std::endl;
+    return false;
+  }
+
+  auto pReparse = reinterpret_cast<PREPARSE_DATA_BUFFER>(GlobalAlloc(GPTR, MAXIMUM_REPARSE_DATA_BUFFER_SIZE));
```

Comment:
`auto` is C++11. Spell the type completely.

Reply:
done